### DISCUSSION TITLE
How to get S parameters matrix data from a solved HFSS design

### DISCUSSION THREAD

**Discussion by AleksandarToshev-eag** [ansys/pyaedt#4973] [Q&A] [answered]
Hi

I am trying to get the S parameters matrix from a solved HFSS design. I am getting FfdSolutionDataExporter object with this:

if setup_of_interest.is_solved:
    ffddata = hfss_instance.get_antenna_ffd_solution_data(
        frequencies=[9e9],
        setup=available_setups[0] + ' : ' + available_sweeps[0],
        sphere="3D")

I was watching some videos demonstration that ffddata should contain touchstone data as well. However I am missing S parameters. Am I using old version of Pyaedt (I guess my version is 0.9.9)? How should I get the S parameters matrix?

**Comment by fwlfHS** [comment, marked as answer]
You could either use `hfss_instance.post.get_solution_data(<...>)` and specify the expression you want to export or directly `hfss_instance.export_touchstone(<...>)` if you like to get a standard touchstone file as output.

**Reply by AleksandarToshev-eag** [reply]
Can I use get_solution_data for getting the full S matrix? I mean if I have a 100 port device in my project I would need 100X100XNum_of_frequencies full S matrix. Can I get this with get_solution_data ?

**Reply by AleksandarToshev-eag** [reply]
I guess if I want the full matrix, than export_touchstone would be more appropriate to use. Unfortunately it will export file :) rather than passing the data directly to other programs

**Reply by AleksandarToshev-eag** [reply]
Thanks for the help!

**Reply by fwlfHS** [reply]
yes. if you leave the `expression` argument empty you get all traces, so all ports. Please check also the examples given in the [documentation](https://aedt.docs.pyansys.com/version/stable/API/_autosummary/pyaedt.modules.AdvancedPostProcessing.PostProcessor.get_solution_data.html#pyaedt.modules.AdvancedPostProcessing.PostProcessor.get_solution_data).

**Reply by AleksandarToshev-eag** [reply]
Thank You!

**Reply by Samuelopez-ansys** [reply]
Hi @AleksandarToshev-eag ,

You can export the touchstone also:
from pyaedt.application.analysis_hf import ScatteringMethods
from pyaedt import Hfss
app = Hfss()
scattering = ScatteringMethods(app)

setup_sweep_parts = setup_name.split(":")

setup_name = setup_sweep_parts[0].strip()
sweep_name = setup_sweep_parts[1].strip()

touchstone_file = scattering.export_touchstone(setup=setup_name, sweep=sweep_name)

This pull request will export Sparamters when you export the farfield as well:
https://github.com/ansys/pyaedt/pull/4893

Next week, it should be merged in a new PyAEDT release.

**Reply by AleksandarToshev-eag** [reply]
Thank you Samuel
